db/write_thread.cc

```diff
@@ -345,6 +345,7 @@ void WriteThread::BeginWriteStall() {
       w->status = Status::Incomplete("Write stall");
       SetState(w, STATE_COMPLETED);
       w = prev->link_older;
+      w->link_newer = (prev == &write_stall_dummy_) ? nullptr : prev;
```

Comment:
What if w is nullptr? Why link_newer cannot be assigned to write_stall_dummy_ when prev is write_stall_dummy_?
I am thinking the following change would be easier to read as the meaning of "w" should not change through the code block.
```
if (prev->link_older != nullptr) {
   prev->link_older->link_newer = prev;
} 
w = prev->link_older;

Reply:
w will not be nullptr, as this function is called by a write group leader and the loop will stop at the first (or last, depending on how you look at it) writer in the write group. But I agree its safer to check for nullptr in case this gets called in other contexts.

link_newer cannot be assigned to write_stall_dummy_ as elsewhere in the code, traversal of link_newer assumes it terminates with a nullptr.

Reply:
If we don't enter the loop because w->write_group == nullptr the last writer before write_stall_dummy_ will have a link_newer pointed to write_stall_dummy_, right?

Reply:
Not necessarily. CreateMissingNewerLinks() may not have had a chance to run.